TelemetryProcessors/SamplingTelemetryProcessor.ts

```diff
@@ -0,0 +1,45 @@
+import ContractsModule = require("../Library/Contracts");
+import { CorrelationContext } from "../AutoCollection/CorrelationContextManager";
+
+/**
+ *  A telemetry processor that handles sampling.
+ */
+export function samplingTelemetryProcessor(envelope: ContractsModule.Contracts.Envelope, contextObjects: { correlationContext: CorrelationContext }): boolean {
+    var samplingPercentage = envelope.sampleRate; // Set for us in Client.getEnvelope
+    var isSampledIn = false;
+
+    if (samplingPercentage === null || samplingPercentage === undefined || samplingPercentage >= 100) {
+        return true;
```

Comment:
The envelope will be prepopulated with the default level of sampling before any telemetry processors. Other telemetry processors can change the envelope sample rate field at will. This telemetry processor will always run last, reading the sampleRate field off of the envelope and deciding to keep or reject the data based off of it